### PR TITLE
[Python3 and Django2.2 upgrade] part 4 - Upgrade to Django 2.2

### PR DESCRIPTION
**Pull request by noliveleger**

### CODE REVIEW

File: docker/sync_static.sh [hunk 1]
```diff
@@ -26,7 +23,5 @@ echo "%	chown -R \"${UWSGI_USER}\" \"${KOBOCAT_SRC_DIR}\""
 echo '%%%%%%%%%%%%%%%%%%%%%%'
 
 echo "Syncing to nginx folder..."
-rsync -aq ${KOBOCAT_SRC_DIR}/onadata/static/* /srv/static/
+rsync -aq --delete --chown=www-data "${KPI_SRC_DIR}/onadata/static/" "${NGINX_STATIC_DIR}/"
```
Comment (jnm) [inline docker/sync_static.sh:26]: :ok_hand: fixed later per https://github.com/kobotoolbox/kobocat/pull/682/#commitcomment-40858338

File: onadata/apps/logger/migrations/0007_add_validate_permission_on_xform.py [hunk 2]
```diff
@@ -11,6 +11,6 @@ class Migration(migrations.Migration):
     operations = [
         migrations.AlterModelOptions(
             name='xform',
-            options={'ordering': ('id_string',), 'verbose_name': 'XForm', 'verbose_name_plural': 'XForms', 'permissions': (('view_xform', 'Can view associated data'), ('report_xform', 'Can make submissions to the form'), ('move_xform', 'Can move form between projects'), ('transfer_xform', 'Can transfer form ownership.'), ('validate_xform', 'Can validate submissions.'))},
+            options={'ordering': ('id_string',), 'verbose_name': 'XForm', 'verbose_name_plural': 'XForms', 'permissions': (('report_xform', 'Can make submissions to the form'), ('move_xform', 'Can move form between projects'), ('transfer_xform', 'Can transfer form ownership.'), ('validate_xform', 'Can validate submissions.'))},
```
Comment (jnm) [inline onadata/apps/logger/migrations/0007_add_validate_permission_on_xform.py:14]: :eyes:

File: onadata/apps/logger/south_migrations/0012_add_permission_view_xform.py [hunk 3]
```diff
@@ -21,7 +21,7 @@ def forwards(self, orm):
             ct = ContentType.objects.get(model='xform', app_label='odk_logger')
             Permission.objects.get(content_type=ct, codename='can_view').delete()
             # add new permission label
-            perm, created = Permission.objects.get_or_create(content_type=ct, codename='view_xform', name='Can view associated data')
+            perm, created = Permission.objects.get_or_create(content_type=ct, codename='view_xform', name='Can view associated data!')
```
Comment (jnm) [inline onadata/apps/logger/south_migrations/0012_add_permission_view_xform.py:24]: :laughing: 
all the south migrations get deleted later anyway :ok_hand:

File: onadata/koboform/redirect_middleware.py [hunk 4]
```diff
@@ -35,11 +35,11 @@ def process_view(self, request, view, args, kwargs):
...
             login_url = koboform.redirect_url(login_url)
-            if view_name is 'login':
+            if view_name == 'login':
                 return HttpResponseRedirect(koboform.login_url(next_kobocat_url='/'))
-            if view_name is 'logout':
+            if view_name == 'logout':
                 return HttpResponseRedirect(koboform.redirect_url('/accounts/logout/'))
-            if view_name is 'FHRegistrationView':
+            if view_name == 'FHRegistrationView':
                 return HttpResponseRedirect(koboform.redirect_url('/accounts/register/'))
```
Comment (jnm) [inline onadata/koboform/redirect_middleware.py:43]: trash?

File: onadata/libs/utils/country_field.py [hunk 5]
```diff
@@ -263,7 +263,7 @@ def __init__(self, *args, **kwargs):
         kwargs.setdefault('maxlength', 2)
         kwargs.setdefault('choices', COUNTRIES)
 
-        super(CountryField, self).__init__(*args, **kwargs)
+        super().__init__(*args, **kwargs)
```
Comment (jnm) [inline onadata/libs/utils/country_field.py:266]: whole class possibly unused?

File: onadata/libs/utils/middleware.py [hunk 6]
```diff
@@ -57,21 +57,6 @@ def process_response(self, request, response):
         return response
 
 
-class BrokenClientMiddleware(MiddlewareMixin):
```
Comment (jnm) [inline onadata/libs/utils/middleware.py:60]: probably not needed, but i'll test ODK Collect
https://www.flowdock.com/app/kobotoolbox/kobo/threads/X5w8Y_PgmFpwMPKR4IOVeL09mR7?filter=search&query=BrokenClientMiddleware

File: onadata/settings/base.py [hunk 7]
```diff
@@ -591,3 +589,7 @@ def skip_suspicious_operations(record):
 
 # The maximum size (in bytes) that an upload will be before it gets streamed to the file system
 FILE_UPLOAD_MAX_MEMORY_SIZE = 10485760
+
+# Monkey Patch PyXForm. @ToDo remove after upgrading to v1.1.0
+logger.removeHandler(logging.NullHandler)
+logger.addHandler(logging.NullHandler())
```
Comment (jnm) [inline onadata/settings/base.py:595]: maybe i should add this to the pyxform upgrade issue

File: onadata/koboform/redirect_middleware.py (hunk 4 above)
Comment (jnm) [reply onadata/koboform/redirect_middleware.py:43]: added to punch list

File: onadata/libs/utils/country_field.py (hunk 5 above)
Comment (jnm) [reply onadata/libs/utils/country_field.py:266]: added to punch list

File: onadata/settings/base.py (hunk 7 above)
Comment (jnm) [reply onadata/settings/base.py:595]: added to punch list

File: onadata/libs/utils/middleware.py (hunk 6 above)
Comment (jnm) [reply onadata/libs/utils/middleware.py:60]: added to punch list

Comment (jnm) [reply onadata/libs/utils/middleware.py:60]: ODK Collect has fixed it: https://github.com/getodk/collect/pull/3269

I tested it with tcpdump and my phone set to Français and saw no evidence of localized strings in date headers :+1:

File: onadata/apps/logger/migrations/0007_add_validate_permission_on_xform.py (hunk 2 above)
Comment (JacquelineMorrissette) [reply onadata/apps/logger/migrations/0007_add_validate_permission_on_xform.py:14]: It doesn't appear that the permission object is removed from the database for existing projects. The `view_xform` permission still exists with the same primary key it has in master

**Pull request merged**